libsolidity/analysis/ViewPureChecker.h

```diff
@@ -59,21 +63,24 @@ class ViewPureChecker: private ASTConstVisitor
 	virtual bool visit(MemberAccess const& _memberAccess) override;
 	virtual void endVisit(MemberAccess const& _memberAccess) override;
 	virtual void endVisit(IndexAccess const& _indexAccess) override;
+	virtual bool visit(ModifierInvocation const& _modifier) override;
 	virtual void endVisit(ModifierInvocation const& _modifier) override;
 	virtual void endVisit(FunctionCall const& _functionCall) override;
 	virtual void endVisit(InlineAssembly const& _inlineAssembly) override;
 
 	/// Called when an element of mutability @a _mutability is encountered.
 	/// Creates appropriate warnings and errors and sets @a m_currentBestMutability.
-	void reportMutability(StateMutability _mutability, SourceLocation const& _location);
+	void reportMutability(StateMutability _mutability, SourceLocation const& _location, SourceLocation const& _nestedLocation = SourceLocation());
 
 	std::vector<std::shared_ptr<ASTNode>> const& m_ast;
 	ErrorReporter& m_errorReporter;
 
 	bool m_errors = false;
-	StateMutability m_currentBestMutability = StateMutability::Payable;
+	MutabilityLocation m_bestMutabilityLocation = MutabilityLocation{StateMutability::Payable, SourceLocation()};
```

Comment:
Judging just from the name, I would think this is only a location. Perhaps `m_currentBestMutabilityAndLocation`? (also rename the struct itself)